buildSrc/src/main/java/org/elasticsearch/gradle/DistributionDownloadPlugin.java

```diff
@@ -112,18 +116,16 @@ private void setupRootDownload(Project rootProject, ElasticsearchDistribution di
         String extractedConfigName = "extracted_" + downloadConfigName;
         final Configuration downloadConfig = configurations.create(downloadConfigName);
         configurations.create(extractedConfigName);
-        Object distroDep = dependencyNotation(rootProject, distribution);
-        rootProject.getDependencies().add(downloadConfigName, distroDep);
+        rootProject.getDependencies().add(downloadConfigName, dependencyNotation(rootProject, distribution));
 
         // add task for extraction, delaying resolving config until runtime
         if (distribution.getType() == Type.ARCHIVE || distribution.getType() == Type.INTEG_TEST_ZIP) {
             Supplier<File> archiveGetter = downloadConfig::getSingleFile;
             String extractDir = rootProject.getBuildDir().toPath().resolve("elasticsearch-distros").resolve(extractedConfigName).toString();
-            TaskProvider<Copy> extractTask = rootProject.getTasks().register(extractTaskName, Copy.class, copyTask -> {
-                copyTask.dependsOn(downloadConfig);
-                copyTask.doFirst(t -> rootProject.delete(extractDir));
-                copyTask.into(extractDir);
-                copyTask.from((Callable<FileTree>)() -> {
+            TaskProvider<Sync> extractTask = rootProject.getTasks().register(extractTaskName, Sync.class, syncTask -> {
```

Comment:
We've confirmed this behaves properly for non-clean builds where UP-TO-DATE information got blown away? To test this run a build that extracts a distro with bundled JDK, blow away the `.gradle` folder in the **project workspace** then run another build.

Reply:
I spoke with Mark offline and will test this directly when adding the first use of the download plugin.